Source/Manifest/manifest-xml.env

```diff
@@ -0,0 +1,261 @@
+#       Create or open for edit xml manifest
+###
+#   - Example:
+#       * _t4dManifestXmlStart -o file              = write start xml manifest in file
+#       * _t4dManifestXmlStart                      = echo start xml manifest
+#
+###
+#   - Note:
+#       <Required>
+#       [Optionnal]
+#       {XXXXXXX}       = Value To Be Changed, if {URL} then replace {URL} with a valid URL
+###
+###### DOCUMENTATION BLOC #####
+########### CODE BLOC #########
+    local _t4dErrorCode=1
+    local _Option="$1"
+    local _File="$2"
+
+    local _Start=$(echo "<?xml version=\"1.0\" encoding=\"UTF-8\"?>\n<manifest>")
+
+    if [[ "$_Option" == "-o" ]]; then
+
+        if [[ ! -e $_File ]] ;then
+            echo $_Start > $_File
+        else 
+            local _res=$(cat $_File | grep -v "</manifest>")
+            echo  $_res > $_File
+        fi  
+    else 
+        echo $_Start
+    fi
+
+###############################
+}
+
+_t4dManifestXmlEnd(){
+# Generated From $Tools4Dev_PATH/Engine/G4d/templates.env
+# v1.8.0
+###### _t4dManifestXmlEnd
+#   - Name:
+#       _t4dManifestXmlEnd
+###
+#   - Synopsis:
+#       add  manifest xml end file
+###
+#   - Example:
+#       * _t4dManifestXmlEnd -o file            = add end to xml manifest end file
+#       * _t4dManifestXmlEnd                    = echo xml end manifest 
+#
+###
+#   - Note:
+#       <Required>
+#       [Optionnal]
+#       {XXXXXXX}       = Value To Be Changed, if {URL} then replace {URL} with a valid URL
+###
+###### DOCUMENTATION BLOC #####
+########### CODE BLOC #########
+    local _t4dErrorCode=1
+    local _Option="$1"
+    local _File="$2"
+
+    local _End=$(echo "</manifest>")
+
+    if [[ "$_Option" == "-o" ]]; then
+        echo $_End >> $_File
+    else 
+        echo $_End
+    fi
+
+###############################
+}
+
+
+_t4dManifestXmlSetLine(){
+# Generated From $Tools4Dev_PATH/Engine/G4d/templates.env
+# v1.8.0
+###### _t4dManifestXmlSetLine
+#   - Name:
+#       _t4dManifestXmlSetLine
+###
+#   - Synopsis:
+#       add line in xml manifest
+###
+#   - Example:
+#       _t4dManifestXmlSetLine -o File -t provider "type=kvm" "time=$(date +%s)"              = {One-Line Description}
+#
+###
+#   - Note:
+#       <Required>
+#       [Optionnal]
+#       {XXXXXXX}       = Value To Be Changed, if {URL} then replace {URL} with a valid URL
+###
+###### DOCUMENTATION BLOC #####
+########### CODE BLOC #########
+
+    local _t4dErrorCode=1
+    local _Out="null"
+    local _Type="null"
+    local _Data="null"
+    local _Res=""
+    local _Arg=""
+
+
+    while [[ "$@" != "" ]]; do
+        case "$1" in
+            -out|-o)
+                _Out="$2"
+                shift
+            ;;
+            -type|-t)
+                _Type="$2"
+                _Res="<${_Type}${_Res}"
+                shift
+            ;;
+            *)
+                _Arg="$1"
+                local _Marker=$(echo $_Arg | cut -d"=" -f1)
+                local _Data=$(echo $_Arg | cut -d"=" -f2)
```

Comment:
what if data is containing `=` ? :)

Reply:
in Xml, it is impossible to use the equal sign, or even special characters, the only way is to use their code, so not possible to find an equal in their data

https://dvteclipse.com/documentation/svlinter/How_to_use_special_characters_in_XML.3F.html